editor/RubyScanner.cpp

```diff
@@ -140,7 +140,7 @@ bool Scanner::didBlockStart()
 
 bool Scanner::didBlockEnd()
 {
-    static const QRegExp regex(QLatin1String("24_"));
+    static const QRegExp regex(QLatin1String("(24|29)_"));
```

Comment:
Don't you support multiple blocks opened in a single line?

What happens with the following code (no parentheses at all)?

    if a == 4; if b == 8
        do_something
      else
        do_something_else
      end
      and_another_thing
    end